### ISSUE TITLE
Handling "Location: http:///" header

### ISSUE THREAD

**Issue by lorien** [urllib3/urllib3#1556]
urllib version: 1.24.1

Code
```python
from urllib3 import PoolManager                                                    
                                                                                   
url = 'http://nerdcast-cdn.jovemnerd.com.br'                                       
pool = PoolManager()                                                               
res = pool.request('GET', url)       
```
raises
```
Traceback (most recent call last):
  File "test3.py", line 5, in <module>
    res = pool.request('GET', url)
  File "/home/web/.local/share/virtualenvs/domscan-CHt7EcBh/lib/python3.5/site-packages/urllib3/request.py", line 68, in request   
    **urlopen_kw)
  File "/home/web/.local/share/virtualenvs/domscan-CHt7EcBh/lib/python3.5/site-packages/urllib3/request.py", line 89, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "/home/web/.local/share/virtualenvs/domscan-CHt7EcBh/lib/python3.5/site-packages/urllib3/poolmanager.py", line 344, in urlopen
    and not conn.is_same_host(redirect_location)):
  File "/home/web/.local/share/virtualenvs/domscan-CHt7EcBh/lib/python3.5/site-packages/urllib3/connectionpool.py", line 436, in is_same_host
    host = _ipv6_host(host, self.scheme)
  File "/home/web/.local/share/virtualenvs/domscan-CHt7EcBh/lib/python3.5/site-packages/urllib3/connectionpool.py", line 892, in _ipv6_host
    if host.startswith('[') and host.endswith(']'):
AttributeError: 'NoneType' object has no attribute 'startswith'
```

Reason is invalid location header: `Location: https:///`
Header is not correct but is it OK to rase AttributerError? Maybe should be something more user-friendly, some urllib3 error?

In my web crawler I want to handle network errors and raise any other errors. In case of AttributeError I can't say is it urllib3 (network) error or some error in my code.

**Comment by sethmlarson** [comment]
Our URL parser is getting overhauled as a part of the 1.25.x release. Likely this behavior will change to raise a `LocationParseError`. If you'd like to verify you can try parsing that URL against #1531.

**Comment by sethmlarson** [comment]
Closed by #1531

**Comment by lorien** [comment]
Yeah. It works:
```
....
urllib3.exceptions.LocationValueError: No host specified.
```
